explainer.md

```diff
@@ -236,30 +242,62 @@ user.
 ---
 ### Proposal
 
-The display locking proposal is intended to improve the script, layout, paint,
-as well as parts of the compositing update phases. In particular, it aims to add
-javascript APIs to allow the developer to lock an element for display. This
-means that the element and its subtree's paint output (ie the output of the
-layout and paint phase) will not change while the lock is acquired. In the
-lock-before-append case, this means that the visual state is empty. What this
-allows is for the user-agent to not have to finish processing the locked
-element’s subtree when processing the layout and paint phase. In other words,
-when requested, the user-agent is free to process parts of the subtree,
-eliminating them from the list of things that have changed (i.e. the dirty
-elements list). Note that this dirty list can be indirectly populated as well
-by, for example, changing CSS properties that affect some elements on the page.
-There are some edge cases to consider here which will be discussed later in this
-document.
+The display locking proposal is intended to **improve the script, layout, paint,
+as well as parts of the compositing update phases**. In particular, it aims to add
+javascript APIs to allow the developer to lock an element for display.
 
-##### Element.displayLock
+With the proposed APIs,
+**web authors can control when not to pay the updating costs for a locked element**,
+and also **request the user-agent to do the updates in a non-janky way**.
+
+### Example code
+
+In this example, we want to add a lot of items to a list in a non-janky way.
+
+```html
+<ul id="itemsList">
+ <div>Item #1</div>
+ <div>Item #2</div>
+ <div>Item #3</div>
+</ul>
+
+<script>
+// Remaining items in the list are in |remainingItems|.
+// We want to add the remaining list items asynchronously.
+remainingItems.forEach(item => {
+  requestIdleCallback((deadline) => {
+    let itemEl = createElementForItem(item);
+    itemEl.displayLock.acquire({ timeout: Infinity, activatable: true });
+    itemsList.appendChild(itemEl);
+    // We can do expensive operations to the items without worrying of the rendering costs.
+    // After we finished all the operations we can trigger a co-operative update & commit.
+    // We might do this in a fancier way by not actually committing the element,
+    // leaving it locked and detecting when to commit by using IntersectionObservers etc,
+    // but that's out of the scope of this example :)
+    doExpensiveOperationsToEl(itemEl).then(element.updateAndCommit);
+  });
+});
+
+// If we need the updated style / layout values of the element, we can do it co-operatively.
+function getOffsetTopForLockedEl(element) {
```

Comment:
I'm a little confused by this function. If the element is locked but none of its ancestors are, then I think this will cause layout to run on the whole doc. Locking the element on makes the layout values of things inside it cooperative, right?

Reply:
Locking an element when it is already in the DOM tree only locks its subtree, not the element.

Reply:
Looks good to me.